### DISCUSSION TITLE
pip install spacy==2.2.2 too slow , that take 40-50mins for this tar

### DISCUSSION THREAD

**Discussion by host452b** [explosion/spaCy#11240] [Help: Installation]
## Your Environment

* Operating System:       ubuntu1804
* Python Version Used:   python3.9/python3.8/python3.7
* spaCy Version Used:    2.2.2
* Environment Information: N/A 

Collecting spacy==2.2.2
  Downloading spacy-2.2.2.tar.gz (5.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.8/5.8 MB 89.3 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: still running...
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting scipy
  Downloading scipy-1.8.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (42.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 MB 52.2 MB/s eta 0:00:00

**Comment by polm** [comment]
Sorry you're having trouble with this. 

spaCy 2.2.2 is older than Python 3.9, so there are no wheels for that Python version. That means you have to build spaCy, and possibly many of its dependencies, from source. For that version, Python 3.8 had wheels, so 3.8 or a slightly older version should install more quickly. 

If possible, you might also want to look at spaCy v3, which was released early last year.

**Reply by polm** [reply]
Hm. I just noticed that while your terminal output above is for Python 3.9, you also list other versions. If you have a slow install on other versions (3.8, 3.7) it might be a different issue. I would recommend you upgrade pip (`pip install -U pip`), and if that doesn't work, then please provide verbose output and we can check it.
